qubes/vm/mix/net.py

```diff
@@ -341,6 +343,49 @@ def on_domain_started_net(self, event, **kwargs):
             except (qubes.exc.QubesException, libvirt.libvirtError):
                 vm.log.warning("Cannot attach network", exc_info=1)
 
+    async def apply_deferred_netvm(self):
+        """Apply deferred netvm changes in case qube could not apply it at the
+        time it was requested."""
+        if getattr(self, "is_preload", False):
+            # Networking of preloaded disposable must be done when it is being
+            # marked as used, so we guarantee that it finishes before the user
+            # can use the disposable.
+            return
+        deferred_from = self.features.get("deferred-netvm-original", None)
+        if deferred_from is None:
+            return
+        oldvalue = None
+        if deferred_from in self.app.domains:
+            oldvalue = self.app.domains[deferred_from]
+        self.fire_event(
+            "property-pre-set:netvm",
+            pre_event=True,
+            name="netvm",
+            newvalue=self.netvm,
+            oldvalue=oldvalue,
+        )
+        if self.netvm:
+            self.fire_event(
+                "property-set:netvm",
+                name="netvm",
+                newvalue=self.netvm,
+                oldvalue=oldvalue,
+            )
+        del self.features["deferred-netvm-original"]
+        # Guarantee that uplink has been concluded (established or absent).
+        await asyncio.wait_for(
+            self.run_service_for_stdio("qubes.WaitForNetworkUplink"),
```

Comment:
I think this service needs to run as root, otherwise it will likely break on minimal templates. You can do it here with `user="root"`, or in the vm with `force-user='root'` in `/etc/qubes/rpc-config/qubes.WaitForNetworkUplink`.